src/counters.ts

```diff
@@ -0,0 +1,694 @@
+import { CountersOptions } from './counters';
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+/**
+ * This module contains all the definitions for Counters decorators and the utility functions and definitions thereof
+*/
+import { Min, Max, IsInt, validateSync, ValidationError, IsDefined, IsString, IsNotEmpty, IsBoolean, Matches } from 'class-validator';
+import { jsonDump, nullNanUndefinedEmptyCoalesce, getBoolean, ProcessInfo, getChildrenTree } from './utils';
+import { writeChartToFile, LineData } from './charts';
+import { promisify } from 'util';
+import path = require('path');
+import fs = require('fs');
+import os = require('os');
+import osu = require('node-os-utils');
+import * as pidusage from 'pidusage';
+import assert = require('assert');
+import { ma as sma, ema } from 'moving-averages';
+import { mean, quantile } from 'simple-statistics';
+const writeFileAsync = promisify(fs.writeFile);
+

```

Comment:
writeFile is already async according to this :

[https://nodejs.org/api/fs.html#fs_fs_writefile_file_data_options_callback](https://nodejs.org/api/fs.html#fs_fs_writefile_file_data_options_callback)

Reply:
perhaps I should rename writeFileAsync to writeFilePromisified, because my understanding is that fs.writeFile would not return a promisy without wrapping it in the promisify call.

Reply:
I could use the calls available in fs_promises but at least when I started writing this we were not on a high enough node version for me to use this. it will be ease swap to use those when we do upgrade.